File: threedollar-api-user/src/main/java/com/depromeet/threedollar/api/user/config/enummodel/EnumMapperConfig.java
package com.depromeet.threedollar.api.user.config.enummodel;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import com.depromeet.threedollar.api.user.service.store.dto.type.UserStoreOrderType;
import com.depromeet.threedollar.common.type.BossStoreFeedbackType;
import com.depromeet.threedollar.common.type.DayOfTheWeek;
import com.depromeet.threedollar.common.utils.EnumMapper;
import com.depromeet.threedollar.domain.mongo.boss.domain.store.BossStoreOpenType;
import com.depromeet.threedollar.domain.rds.user.domain.advertisement.AdvertisementPlatformType;
import com.depromeet.threedollar.domain.rds.user.domain.advertisement.AdvertisementPositionType;
import com.depromeet.threedollar.domain.rds.user.domain.faq.FaqCategory;
import com.depromeet.threedollar.domain.rds.user.domain.store.DeleteReasonType;
import com.depromeet.threedollar.common.type.MenuCategoryType;
import com.depromeet.threedollar.domain.rds.user.domain.store.PaymentMethodType;
import com.depromeet.threedollar.domain.rds.user.domain.store.StoreType;
import com.depromeet.threedollar.domain.rds.user.domain.user.UserSocialType;
import com.depromeet.threedollar.domain.rds.user.domain.visit.VisitType;

@Configuration
public class EnumMapperConfig {

    @Bean
    public EnumMapper enumMapper() {
        EnumMapper enumMapper = new EnumMapper();
        // common
        enumMapper.put("DayOfTheWeek", DayOfTheWeek.class);

        // user
        enumMapper.put("UserSocialType", UserSocialType.class);

        // store
        enumMapper.put("PaymentMethodType", PaymentMethodType.class);
        enumMapper.put("StoreType", StoreType.class);
        enumMapper.put("DeleteReasonType", DeleteReasonType.class);
        enumMapper.put("StoreOrderType", UserStoreOrderType.class);
        enumMapper.put("MenuCategoryType", MenuCategoryType.class);

        // faq
        enumMapper.put("FaqCategory", FaqCategory.class);

        // advertisement
        enumMapper.put("AdvertisementPositionType", AdvertisementPositionType.class);
        enumMapper.put("AdvertisementPlatformType", AdvertisementPlatformType.class);

        // visit
        enumMapper.put("VisitType", VisitType.class);

        // bossStore
        enumMapper.put("BossStoreOpenType", BossStoreOpenType.class);
        enumMapper.put("BossStoreFeedbackType", BossStoreFeedbackType.class);

        return enumMapper;
    }

}

File: threedollar-api-admin/src/main/kotlin/com/depromeet/threedollar/api/admin/config/enummapper/EnumMapperConfig.kt
package com.depromeet.threedollar.api.admin.config.enummapper

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import com.depromeet.threedollar.common.type.DayOfTheWeek
import com.depromeet.threedollar.common.utils.EnumMapper
import com.depromeet.threedollar.domain.rds.vendor.domain.advertisement.AdvertisementPlatformType
import com.depromeet.threedollar.domain.rds.vendor.domain.advertisement.AdvertisementPositionType
import com.depromeet.threedollar.domain.rds.vendor.domain.faq.FaqCategory
import com.depromeet.threedollar.domain.rds.vendor.domain.store.DeleteReasonType
import com.depromeet.threedollar.domain.rds.vendor.domain.store.PaymentMethodType
import com.depromeet.threedollar.domain.rds.vendor.domain.store.StoreType
import com.depromeet.threedollar.domain.rds.vendor.domain.user.UserSocialType

@Configuration
class EnumMapperConfig {

    @Bean
    fun enumMapper(): EnumMapper {
        val enumMapper = EnumMapper()
        // common
        enumMapper.put("DayOfTheWeek", DayOfTheWeek::class.java)

        // user
        enumMapper.put("UserSocialType", UserSocialType::class.java)

        // store
        enumMapper.put("PaymentMethodType", PaymentMethodType::class.java)
        enumMapper.put("StoreType", StoreType::class.java)
        enumMapper.put("DeleteReasonType", DeleteReasonType::class.java)

        // faq
        enumMapper.put("FaqCategory", FaqCategory::class.java)

        // advertisement
        enumMapper.put("AdvertisementPositionType", AdvertisementPositionType::class.java)
        enumMapper.put("AdvertisementPlatformType", AdvertisementPlatformType::class.java)

        return enumMapper
    }

}
